tools/vscode/src/lib.rs

```diff
@@ -0,0 +1,538 @@
+        Self {
+            workspace_root,
+            bazel_binary: "bazel".into(),
+            bazel_info,
+        }
+    }
+
+    pub fn with_bazel_binary(mut self, bazel: Utf8PathBuf) -> Self {
+        self.bazel_binary = bazel;
+        self
+    }
+
+    /// Query information about multiple targets at once.
+    pub fn query_targets_batch(&mut self, targets: &[String]) -> Result<Vec<TargetInfo>> {
+        if targets.is_empty() {
+            return Ok(vec![]);
+        }
+
+        // Get all target kinds in one query
+        let target_kinds = self.batch_query_target_kinds(targets)?;
+
+        let mut results = Vec::new();
+        for target in targets {
+            if let Some(target_kind) = target_kinds.get(target) {
+                let is_test = target_kind.contains("rust_test");
+
+                // We don't need to resolve binary paths now - that will happen at debug time
+                // Just create a placeholder path that will be resolved by the pre-launch task
+                results.push(TargetInfo {
+                    label: target.to_string(),
+                    binary_path: Utf8PathBuf::from(""), // Placeholder - resolved at debug time
+                    is_test,
+                    target_kind: target_kind.clone(),
+                });
+            }
+        }
+
+        Ok(results)
+    }
+
+    /// Query information about a specific target.
+    pub fn query_target_info(&mut self, target: &str) -> Result<TargetInfo> {
+        let results = self.query_targets_batch(&[target.to_string()])?;
+        results
+            .into_iter()
+            .next()
+            .ok_or_else(|| anyhow::anyhow!("Failed to query target {}", target))
+    }
+
+    /// Batch query target kinds for multiple targets at once.
+    fn batch_query_target_kinds(&self, targets: &[String]) -> Result<BTreeMap<String, String>> {
+        let target_pattern = targets.join(" + ");
+
+        let output = bazel_command(&self.bazel_binary, &self.workspace_root)
+            .arg("query")
+            .arg("--output=label_kind")
+            .arg(&target_pattern)
+            .output()
+            .context("Failed to execute 'bazel query'")?;
+
+        if !output.status.success() {
+            let stderr = String::from_utf8_lossy(&output.stderr);
+            bail!("bazel query failed for targets: {}", stderr);
+        }
+
+        let stdout = String::from_utf8(output.stdout)?;
+        let mut result = BTreeMap::new();
+
+        for line in stdout.trim().lines() {
+            // Format: "rust_test rule //path/to:target"
+            let parts: Vec<&str> = line.split_whitespace().collect();
+            if parts.len() >= 3 {
+                let kind = parts[0];
+                let label = parts[2];
+
+                if !kind.starts_with("rust_") {
+                    // Skip non-Rust targets
+                    continue;
+                }
+
+                result.insert(label.to_string(), line.to_string());
```

Comment:
```suggestion
                result.insert(label.to_string(), kind.to_string());
```

Right now you're saving the whole line, not just the kind, which you can get away with because you happen to only be doing `contains` checks